### PR TITLE
ci: add go mod linter to lint target

### PR DESCRIPTION
**Pull request by Xunzhuo**
Signed-off-by: bitliu <bitliu@tencent.com>

After #743.

### CODE REVIEW

**Comment by danehans** [comment]
@Xunzhuo #743 has merged so feel free to update this PR.

File: tools/make/golang.mk [hunk 1]
```diff
@@ -78,8 +80,8 @@ test: ## Run all Go test of code sources.
 test: go.test.unit
 
 .PHONY: format
-format: ## Update dependences with mod tidy.
-format: go.tidy
+format: ## Update and check dependences with go mod tidy.
+format: go.mod.lint
```
Comment (danehans) [inline tools/make/golang.mk:84]: Why do we need the `format` target if it's doing the same thing as the `lint` target?

Comment (Xunzhuo) [reply tools/make/golang.mk:84]: `format` just does the go mod tidy locally, but `lint` includes other linter checks. We can add more format things like goimport under format, which can format our codes.

**Comment by danehans** [comment]
@Xunzhuo LGTM after a rebase.

**Review by arkodg — APPROVED** [review]

**Pull request merged**